pkgs/tools/networking/pihole-ftl/default.nix

```diff
@@ -0,0 +1,50 @@
+{ cmake
+, fetchFromGitHub
+, gmp
+, lib
+, libidn
+, nettle
+, readline
+, stdenv
+}:
+
+stdenv.mkDerivation rec {
+  pname = "pihole-ftl";
+  version = "5.7";
```

Comment:
Also, 5.8.1 was recently released :)

Reply:
I bumped the version, but it required an extra patch. Upstream added the `-Werror` flag, but it fails to compile with GCC 9+.

Reply:
I think there's a hardening thing we can disable so we don't have to patch the source... I'll see if I can find it and suggest it once I do.

Reply:
The manual doesn't say anything about hardening flags that can disable werror.

Reply:
I was thinking it would be the `format` hardening, but it wasn't. Here's the diff I applied for this:

```diff
diff --git a/pkgs/tools/networking/pihole-ftl/default.nix b/pkgs/tools/networking/pihole-ftl/default.nix
index da109e2181a..7a969fa3272 100644
--- a/pkgs/tools/networking/pihole-ftl/default.nix
+++ b/pkgs/tools/networking/pihole-ftl/default.nix
@@ -23,10 +23,14 @@ stdenv.mkDerivation rec {
 
   buildInputs = [ gmp libidn nettle readline ];
 
+  NIX_CFLAGS_COMPILE = [
+    # Upstream converts from enum to const enum
+    "-Wno-error=enum-conversion"
+  ];
+
   postPatch = ''
     substituteInPlace src/CMakeLists.txt \
-      --replace "CMAKE_STATIC_LIBRARY_SUFFIX" "CMAKE_SHARED_LIBRARY_SUFFIX" \
-      --replace "EXTRAWARN \"-Werror -Waddress" "EXTRAWARN \"-Waddress"
+      --replace "CMAKE_STATIC_LIBRARY_SUFFIX" "CMAKE_SHARED_LIBRARY_SUFFIX"
 
     substituteInPlace src/version.h.in \
       --replace "@GIT_VERSION@" "v${version}" \
```

Reply:
I prefer disabling werror to prevent any new warnings in future compilers breaking the build. You never know what new warnings are added.

Reply:
The above diff isn't quite right, since apparently we don't want to use `NIX_CFLAGS_COMPILE`: https://github.com/NixOS/nixpkgs/issues/79303. Changing it to just `CFLAGS` works fine as well, though. (I've edited the diff to make this change.)

Reply:
I've fixed this with `CFLAGS` for now. It's been [fixed upstream](https://github.com/pi-hole/FTL/pull/1120) and this workaround should be removed when they cut a new release.